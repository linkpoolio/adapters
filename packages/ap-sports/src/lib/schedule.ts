import { datetime, errors, uuid } from '@linkpool/shared'

import { statusToStatusId } from './constants'
import type { Game, GameCreate, GameResolve } from './types'

export function getGameCreate(game: Game): GameCreate {
  const { home, away } = game
  const gameCreate = {
    gameId: uuid.uuidToBytes32(game.id),
    startTime: datetime.iso8061ToTimestamp(game.scheduled),
    homeTeam: `${home.market} ${home.name}`,
    awayTeam: `${away.market} ${away.name}`,
  }
  Object.entries(gameCreate).forEach(([key, value]) => {
    value ??
      errors.throwError(
        `'${key}' is null or undefined: ${JSON.stringify(gameCreate)}. Game: ${JSON.stringify(
          game,
        )}`,
      )
  })
  return gameCreate
}

export function getGameResolve(game: Game): GameResolve {
  const { home, away } = game
  const gameResolve = {
    gameId: uuid.uuidToBytes32(game.id),
    homeScore: home.runs,
    awayScore: away.runs,
    status: statusToStatusId.get(game.status) as number,
  }
  Object.entries(gameResolve).forEach(([key, value]) => {
    value ??
      errors.throwError(
        `'${key}' is null or undefined: ${JSON.stringify(gameResolve)}. Game: ${JSON.stringify(
          game,
        )}`,
      )
  })
  return gameResolve
}

export function stringToHex(id: string): string {
  const bytes = Buffer.from(id.replace(/-/g, ''))
  if (bytes.length === 32) {
    return `0x${bytes.toString('hex')}`
  }
  throw new Error(`Unexpected 'id': ${id}. The data type bytes32 expected length is 32.`)
}
