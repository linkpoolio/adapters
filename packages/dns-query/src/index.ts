import { expose } from '@chainlink/ea-bootstrap'

import { makeExecute } from './adapter'
import { NAME, makeConfig } from './config'
import rateLimit from './config/limits.json'

const adapterContext = { name: NAME, rateLimit }

const { server } = expose(adapterContext, makeExecute())
export { NAME, makeConfig, makeExecute, server }
