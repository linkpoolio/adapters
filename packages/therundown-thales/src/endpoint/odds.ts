import { AdapterError, Requester, Validator, util } from '@chainlink/ea-bootstrap'
import type { Config, ExecuteWithConfig, InputParameters } from '@chainlink/types'
import { join } from 'path'

import { Market, marketToStatus, maxLimit, supportedSportIdOdds } from '../lib/const'
import type { Event, GameOdds } from '../lib/types'
import {
  encodeGameOdds,
  filterByEventId,
  filterEventStatus,
  getGameOdds,
  validateAndGetGameIds,
  validateDate,
  validateSportId,
} from '../lib/utils'

export const supportedEndpoints = ['odds']

export const inputParameters: InputParameters = {
  sportId: {
    description: 'The ID of the sport to query',
    required: true,
    type: 'number',
    options: supportedSportIdOdds,
  },
  date: {
    description: 'The date of the games to query as a Unix timestamp seconds.',
    required: true,
    type: 'number',
  },
  gameIds: {
    description:
      'The IDs of games to query. Example: `["23660869053591173981da79133fe4c2","fb78cede8c9aa942b2569b048e649a3f"]`',
    required: false,
  },
}

export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
  const validator = new Validator(request, inputParameters)

  const jobRunID = validator.validated.id
  const sportId = validator.validated.data.sportId
  const dateRaw = validator.validated.data.date
  const gameIdsRaw = validator.validated.data.gameIds

  let gameIds: string[] = []
  let date: string
  try {
    validateSportId(sportId)
    date = validateDate(dateRaw)
    gameIds = validateAndGetGameIds(gameIdsRaw)
  } catch (error) {
    const message = (error as Error).message
    throw new AdapterError({
      jobRunID,
      statusCode: 400,
      message,
      cause: error,
    })
  }

  const url = util.buildUrlPath('sports/:sportId/events/:date?include=:include&limit=:limit', {
    sportId,
    date,
    include: 'scores',
    limit: maxLimit,
  })

  const reqConfig = {
    ...config.api,
    headers: {
      ...config.api.headers,
      'x-rapidapi-key': config.apiKey,
    },

    url,
  }

  const response = await Requester.request(reqConfig)

  const events: Event[] = response.data.events

  if (!Array.isArray(events)) {
    throw new AdapterError({
      jobRunID,
      statusCode: 500,
      message: `Unexpected 'events' format in data: ${events}. Expected array.`,
      url: join(reqConfig.baseURL, reqConfig.url),
    })
  }
  if (events.length === 0) {
    response.data.result = []

    return Requester.success(jobRunID, response, config.verbose)
  }

  const statuses = marketToStatus.get(Market.CREATE) as string[]

  const filteredEvents = events.filter((event: Event) => {
    return filterByEventId(event, gameIds) && filterEventStatus(event, statuses)
  })
  let oddsList: GameOdds[]
  let encodedOddsList: string[]
  try {
    oddsList = filteredEvents.map((event: Event) => getGameOdds(event))
    encodedOddsList = oddsList.map((gameOdds: GameOdds) => encodeGameOdds(gameOdds))
  } catch (error) {
    const message = (error as Error).message
    throw new AdapterError({
      jobRunID,
      statusCode: 500,
      message,
      cause: error,
      url: join(reqConfig.baseURL, reqConfig.url),
    })
  }
  response.data.result = encodedOddsList

  return Requester.success(jobRunID, response, config.verbose)
}
