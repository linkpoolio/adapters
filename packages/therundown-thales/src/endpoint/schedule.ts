import { AdapterError, Requester, Validator, util } from '@chainlink/ea-bootstrap'
import type { Config, ExecuteWithConfig, InputParameters } from '@chainlink/types'
import { join } from 'path'

import { Market, marketToStatus, maxLimit, supportedSportIdSchedule } from '../lib/const'
import type { Event, GameCreate, GameResolve } from '../lib/types'
import {
  encodeGameCreate,
  encodeGameResolve,
  filterByEventId,
  filterEventStatus,
  getGameCreate,
  getGameResolve,
  validateAndGetGameIds,
  validateAndGetStatusIds,
  validateDate,
  validateMarket,
  validateSportId,
} from '../lib/utils'

export const supportedEndpoints = ['schedule']

export const inputParameters: InputParameters = {
  sportId: {
    description: 'The ID of the sport to query',
    required: true,
    type: 'number',
    options: supportedSportIdSchedule,
  },
  date: {
    description: 'The date of the games to query as a Unix timestamp seconds.',
    required: true,
    type: 'number',
  },
  market: {
    description: 'Chose to create or resolve market',
    required: true,
    type: 'string',
    options: [Market.CREATE, Market.RESOLVE],
  },
  statusIds: {
    description: 'The statuses of the games to query. Examples: `["1","2","3"]`',
    required: false,
  },
  gameIds: {
    description:
      'The IDs of games to query. Example: `["23660869053591173981da79133fe4c2","fb78cede8c9aa942b2569b048e649a3f"]`',
    required: false,
  },
}

export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
  const validator = new Validator(request, inputParameters)

  const jobRunID = validator.validated.id
  const sportId = validator.validated.data.sportId
  const dateRaw = validator.validated.data.date
  const market = validator.validated.data.market
  const gameIdsRaw = validator.validated.data.gameIds
  const statusIdsRaw = validator.validated.data.statusIds

  let gameIds: string[] = []
  let statusIds: string[] = []
  let date: string
  try {
    validateMarket(market as Market)
    validateSportId(sportId)
    date = validateDate(dateRaw)
    gameIds = validateAndGetGameIds(gameIdsRaw)
    statusIds = validateAndGetStatusIds(statusIdsRaw)
  } catch (error) {
    const message = (error as Error).message
    throw new AdapterError({
      jobRunID,
      statusCode: 400,
      message,
      cause: error,
    })
  }

  const url = util.buildUrlPath('sports/:sportId/events/:date?include=:include&limit=:limit', {
    sportId,
    date,
    include: 'scores',
    limit: maxLimit,
  })

  const reqConfig = {
    ...config.api,
    headers: {
      ...config.api.headers,
      'x-rapidapi-key': config.apiKey,
    },

    url,
  }

  const response = await Requester.request(reqConfig)

  const events: Event[] = response.data.events

  if (!Array.isArray(events)) {
    throw new AdapterError({
      jobRunID,
      statusCode: 200,
      message: `Unexpected 'events' format in data: ${events}. Expected array.`,
      url: join(reqConfig.baseURL, reqConfig.url),
    })
  }
  if (events.length === 0) {
    response.data.result = []

    return Requester.success(jobRunID, response, config.verbose)
  }

  if (market === Market.CREATE) {
    const filteredEvents = events.filter((event: Event) => {
      return (marketToStatus.get(Market.CREATE) as string[]).includes(event.score?.event_status)
    })
    let gameCreateList: GameCreate[]
    let encodedGameCreateList: string[]
    try {
      gameCreateList = filteredEvents.map((event: Event) => getGameCreate(event, sportId))
      encodedGameCreateList = gameCreateList.map((gameCreate: GameCreate) =>
        encodeGameCreate(gameCreate),
      )
    } catch (error) {
      const message = (error as Error).message
      throw new AdapterError({
        jobRunID,
        statusCode: 200,
        message,
        cause: error,
        url: join(reqConfig.baseURL, reqConfig.url),
      })
    }
    response.data.result = encodedGameCreateList

    return Requester.success(jobRunID, response, config.verbose)
  }

  if (market === Market.RESOLVE) {
    let statuses: string[]
    if (statusIds.length !== 0) {
      statuses = statusIds.filter(
        (statusId: string) => !(marketToStatus.get(Market.CREATE) as string[]).includes(statusId),
      )
    } else {
      statuses = marketToStatus.get(Market.RESOLVE) as string[]
    }

    const filteredEvents = events.filter((event: Event) => {
      return filterByEventId(event, gameIds) && filterEventStatus(event, statuses)
    })
    let gameResolveList: GameResolve[]
    let encodedGameResolveList: string[]
    try {
      gameResolveList = filteredEvents.map((event: Event) => getGameResolve(event, sportId))
      encodedGameResolveList = gameResolveList.map((gameResolve: GameResolve) =>
        encodeGameResolve(gameResolve),
      )
    } catch (error) {
      const message = (error as Error).message
      throw new AdapterError({
        jobRunID,
        statusCode: 200,
        message,
        cause: error,
        url: join(reqConfig.baseURL, reqConfig.url),
      })
    }
    response.data.result = encodedGameResolveList

    return Requester.success(jobRunID, response, config.verbose)
  }

  throw new Error(`Unsupported market: ${market}`)
}
