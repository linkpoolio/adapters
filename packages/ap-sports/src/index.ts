import { expose } from '@chainlink/ea-bootstrap'

import { endpointSelector, makeExecute } from './adapter'
import { NAME, makeConfig } from './config'
import * as rateLimit from './config/limits.json'
import * as endpoints from './endpoint'
import type * as types from './lib/types'

const adapterContext = { name: NAME, rateLimit }

const { server } = expose(adapterContext, makeExecute(), undefined, endpointSelector)
export { NAME, makeExecute, makeConfig, server, types, endpoints }
