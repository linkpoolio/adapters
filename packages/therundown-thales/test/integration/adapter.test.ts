import http from 'http'
import type { AddressInfo } from 'net'
import nock from 'nock'
import request from 'supertest'
import type { SuperTest, Test } from 'supertest'

import { server as startServer } from '../../src'
import { oddsTests } from './odds'
import { scheduleTests } from './schedule'

export interface SuiteContext {
  server: http.Server
  req: SuperTest<Test>
}

describe('execute', () => {
  const context: SuiteContext = {
    server: null,
    req: null,
  }

  beforeAll(async () => {
    process.env.API_KEY = process.env.API_KEY || 'fake-api-key'
    process.env.CACHE_ENABLED = 'false'
    if (process.env.RECORD) {
      nock.recorder.rec()
    }
  })

  afterAll(() => {
    if (process.env.RECORD) {
      nock.recorder.play()
    }

    nock.restore()
    nock.cleanAll()
    nock.enableNetConnect()
  })

  beforeEach(async () => {
    context.server = await startServer()
    context.req = request(`localhost:${(context.server.address() as AddressInfo).port}`)
  })

  afterEach((done) => {
    context.server.close(done)
  })

  describe('shcedule endpoint', () => scheduleTests(context))
  describe('odds endpoint', () => oddsTests(context))
})
