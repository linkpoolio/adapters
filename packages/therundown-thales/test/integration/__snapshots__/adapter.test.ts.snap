// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute odds endpoint error calls when successfully requesting Therundown API should throw and exception if the response is malformed 1`] = `
Object {
  "error": Object {
    "feedID": "{\\"data\\":{\\"endpoint\\":\\"odds\\",\\"sportId\\":4,\\"date\\":1635529231,\\"sportIdToBookmakerIds\\":{\\"4\\":[11,3]}}}",
    "message": "The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received type number (777)",
    "name": "AdapterError",
    "url": "https:/therundown-therundown-v1.p.rapidapi.com/sports/4/events/2021-10-29?include=scores&limit=500",
  },
  "jobRunID": "1",
  "status": "errored",
  "statusCode": 500,
}
`;

exports[`execute odds endpoint success calls when successfully requesting Therundown API should return 1 result 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x36386530393232306339666237646239373035353530666165386138383332320000000000000000000000000000000000000000000000000000000000015ae00000000000000000000000000000000000000000000000000000000000012f840000000000000000000000000000000000000000000000000000000000000000",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x36386530393232306339666237646239373035353530666165386138383332320000000000000000000000000000000000000000000000000000000000015ae00000000000000000000000000000000000000000000000000000000000012f840000000000000000000000000000000000000000000000000000000000000000",
  ],
  "statusCode": 200,
}
`;

exports[`execute odds endpoint success calls when successfully requesting Therundown API should return 2 results (case filtering by gameIds) 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x3030313730343961333736636439633733333435353037373637323935633734000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
      "0x3033613234326133343661363338333564396261313739376633613130666638000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x3030313730343961333736636439633733333435353037373637323935633734000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
    "0x3033613234326133343661363338333564396261313739376633613130666638000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
  ],
  "statusCode": 200,
}
`;

exports[`execute odds endpoint success calls when successfully requesting Therundown API should return 2 results (case filtering by gameIds) 2`] = `
Object {
  "data": Object {
    "result": Array [],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint error calls when successfully requesting Therundown API should throw an exception if a game attribute is malformed while creating market 1`] = `
Object {
  "error": Object {
    "feedID": "{\\"data\\":{\\"endpoint\\":\\"schedule\\",\\"sportId\\":4,\\"date\\":1635529231,\\"market\\":\\"create\\",\\"sportIdToBookmakerIds\\":{\\"4\\":[3,11]}}}",
    "message": "Missing 'teams_normalized' in event: {\\"event_id\\":777,\\"event_uuid\\":\\"11ec-384b-28cf0000-823d-cd821c1e16bb\\",\\"score\\":{\\"event_status\\":\\"STATUS_SCHEDULED\\",\\"score_away\\":23,\\"score_home\\":91,\\"winner_away\\":1,\\"winner_home\\":0},\\"teams\\":[{\\"team_id\\":152,\\"team_normalized_id\\":20,\\"is_away\\":true,\\"is_home\\":false},{\\"team_id\\":188,\\"team_normalized_id\\":27,\\"name\\":\\"Houston Rockets\\",\\"is_away\\":false,\\"is_home\\":true}]}",
    "name": "AdapterError",
    "url": "https:/therundown-therundown-v1.p.rapidapi.com/sports/4/events/2021-10-29?include=scores&limit=500",
  },
  "jobRunID": "1",
  "status": "errored",
  "statusCode": 500,
}
`;

exports[`execute schedule endpoint error calls when successfully requesting Therundown API should throw an exception if a game attribute is malformed while resolving market 1`] = `
Object {
  "error": Object {
    "feedID": "{\\"data\\":{\\"endpoint\\":\\"schedule\\",\\"sportId\\":4,\\"date\\":1635529231,\\"market\\":\\"resolve\\",\\"sportIdToBookmakerIds\\":{\\"4\\":[3,11]}}}",
    "message": "'awayScore' is null or undefined: {\\"homeScore\\":91,\\"awayScore\\":null,\\"gameId\\":\\"0x3432656630396566643762313465663961393664363831363738356236316265\\",\\"statusId\\":8,\\"updatedAt\\":null}. Event: {\\"event_id\\":\\"42ef09efd7b14ef9a96d6816785b61be\\",\\"event_uuid\\":\\"11ec-384b-28cf0000-823d-cd821c1e16bb\\",\\"score\\":{\\"event_status\\":\\"STATUS_FINAL\\",\\"score_away\\":null,\\"score_home\\":91,\\"winner_away\\":1,\\"winner_home\\":0},\\"teams\\":[{\\"team_id\\":152,\\"team_normalized_id\\":20,\\"is_away\\":true,\\"is_home\\":false},{\\"team_id\\":188,\\"team_normalized_id\\":27,\\"name\\":\\"Houston Rockets\\",\\"is_away\\":false,\\"is_home\\":true}]}",
    "name": "AdapterError",
    "url": "https:/therundown-therundown-v1.p.rapidapi.com/sports/4/events/2021-10-29?include=scores&limit=500",
  },
  "jobRunID": "1",
  "status": "errored",
  "statusCode": 500,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 0 results (case create market filtering by gameIds) 1`] = `
Object {
  "data": Object {
    "result": Array [],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 0 results (case resolve market filtering by gameIds) 1`] = `
Object {
  "data": Object {
    "result": Array [],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 1 result (case create market) 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x0000000000000000000000000000000000000000000000000000000000000020363865303932323063396662376462393730353535306661653861383833323200000000000000000000000000000000000000000000000000000000624e2980ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff2734000000000000000000000000000000000000000000000000000000000000b220000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000d41746c616e7461204861776b7300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001257617368696e67746f6e2057697a617264730000000000000000000000000000",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x0000000000000000000000000000000000000000000000000000000000000020363865303932323063396662376462393730353535306661653861383833323200000000000000000000000000000000000000000000000000000000624e2980ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff2734000000000000000000000000000000000000000000000000000000000000b220000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000d41746c616e7461204861776b7300000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001257617368696e67746f6e2057697a617264730000000000000000000000000000",
  ],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 1 result (case resolve market) 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x3464373363633831373762393464346439653230353266613537656263663163000000000000000000000000000000000000000000000000000000000000005b000000000000000000000000000000000000000000000000000000000000007a00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000062db3e25",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x3464373363633831373762393464346439653230353266613537656263663163000000000000000000000000000000000000000000000000000000000000005b000000000000000000000000000000000000000000000000000000000000007a00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000062db3e25",
  ],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 2 results (case create market filtering by gameIds) 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x0000000000000000000000000000000000000000000000000000000000000020303031373034396133373663643963373333343535303737363732393563373400000000000000000000000000000000000000000000000000000000631cb48000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000000134f68696f205374617465204275636b6579657300000000000000000000000000000000000000000000000000000000000000000000000000000000000000001941726b616e7361732053746174652052656420576f6c76657300000000000000",
      "0x0000000000000000000000000000000000000000000000000000000000000020303361323432613334366136333833356439626131373937663361313066663800000000000000000000000000000000000000000000000000000000631cb48000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000e416c62696f6e20427269746f6e730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010426c756666746f6e204265617665727300000000000000000000000000000000",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x0000000000000000000000000000000000000000000000000000000000000020303031373034396133373663643963373333343535303737363732393563373400000000000000000000000000000000000000000000000000000000631cb48000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000000134f68696f205374617465204275636b6579657300000000000000000000000000000000000000000000000000000000000000000000000000000000000000001941726b616e7361732053746174652052656420576f6c76657300000000000000",
    "0x0000000000000000000000000000000000000000000000000000000000000020303361323432613334366136333833356439626131373937663361313066663800000000000000000000000000000000000000000000000000000000631cb48000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000e416c62696f6e20427269746f6e730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010426c756666746f6e204265617665727300000000000000000000000000000000",
  ],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 2 results (case resolve market filtering by gameIds) 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x33393235343665313435303739643064336433323832623430373564373132370000000000000000000000000000000000000000000000000000000000000038000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000063149f39",
      "0x3034303236356364633130323265313365663137363462396137326363613433000000000000000000000000000000000000000000000000000000000000000700000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000006313be0a",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x33393235343665313435303739643064336433323832623430373564373132370000000000000000000000000000000000000000000000000000000000000038000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000063149f39",
    "0x3034303236356364633130323265313365663137363462396137326363613433000000000000000000000000000000000000000000000000000000000000000700000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000006313be0a",
  ],
  "statusCode": 200,
}
`;

exports[`execute schedule endpoint success calls when successfully requesting Therundown API should return 6 results (case create market with no "lines") 1`] = `
Object {
  "data": Object {
    "result": Array [
      "0x000000000000000000000000000000000000000000000000000000000000002036343764366262646332613765333335323966653031303666626234373734380000000000000000000000000000000000000000000000000000000062cc677800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000000124b616e736173204369747920526f79616c730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e446574726f697420546967657273000000000000000000000000000000000000",
      "0x000000000000000000000000000000000000000000000000000000000000002031313463333262666531396134643133623435303761336265653266383735610000000000000000000000000000000000000000000000000000000062cca6c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000d4d69616d69204d61726c696e730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000125069747473627572676820506972617465730000000000000000000000000000",
      "0x000000000000000000000000000000000000000000000000000000000000002038386461383262633133353135376661656132313963656438653461346333380000000000000000000000000000000000000000000000000000000062ccadc800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000e54616d7061204261792052617973000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e426f73746f6e2052656420536f78000000000000000000000000000000000000",
      "0x000000000000000000000000000000000000000000000000000000000000002062343466646664383132373738376132363061383835653639666635666134340000000000000000000000000000000000000000000000000000000062ccadc800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e000000000000000000000000000000000000000000000000000000000000001200000000000000000000000000000000000000000000000000000000000000013436c6576656c616e6420477561726469616e730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000114368696361676f20576869746520536f78000000000000000000000000000000",
      "0x000000000000000000000000000000000000000000000000000000000000002061373236653863333261656434336534386635363933623162383731306332640000000000000000000000000000000000000000000000000000000062ccaef400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000001353742e204c6f7569732043617264696e616c730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000155068696c6164656c70686961205068696c6c6965730000000000000000000000",
      "0x000000000000000000000000000000000000000000000000000000000000002038666462366537396630343733626366393962666532643331343831623261380000000000000000000000000000000000000000000000000000000062ccb02000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000e41746c616e746120427261766573000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000d4e657720596f726b204d65747300000000000000000000000000000000000000",
    ],
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": Array [
    "0x000000000000000000000000000000000000000000000000000000000000002036343764366262646332613765333335323966653031303666626234373734380000000000000000000000000000000000000000000000000000000062cc677800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000012000000000000000000000000000000000000000000000000000000000000000124b616e736173204369747920526f79616c730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e446574726f697420546967657273000000000000000000000000000000000000",
    "0x000000000000000000000000000000000000000000000000000000000000002031313463333262666531396134643133623435303761336265653266383735610000000000000000000000000000000000000000000000000000000062cca6c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000d4d69616d69204d61726c696e730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000125069747473627572676820506972617465730000000000000000000000000000",
    "0x000000000000000000000000000000000000000000000000000000000000002038386461383262633133353135376661656132313963656438653461346333380000000000000000000000000000000000000000000000000000000062ccadc800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000e54616d7061204261792052617973000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e426f73746f6e2052656420536f78000000000000000000000000000000000000",
    "0x000000000000000000000000000000000000000000000000000000000000002062343466646664383132373738376132363061383835653639666635666134340000000000000000000000000000000000000000000000000000000062ccadc800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e000000000000000000000000000000000000000000000000000000000000001200000000000000000000000000000000000000000000000000000000000000013436c6576656c616e6420477561726469616e730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000114368696361676f20576869746520536f78000000000000000000000000000000",
    "0x000000000000000000000000000000000000000000000000000000000000002061373236653863333261656434336534386635363933623162383731306332640000000000000000000000000000000000000000000000000000000062ccaef400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000001353742e204c6f7569732043617264696e616c730000000000000000000000000000000000000000000000000000000000000000000000000000000000000000155068696c6164656c70686961205068696c6c6965730000000000000000000000",
    "0x000000000000000000000000000000000000000000000000000000000000002038666462366537396630343733626366393962666532643331343831623261380000000000000000000000000000000000000000000000000000000062ccb02000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000000000000000000000000000000000000000120000000000000000000000000000000000000000000000000000000000000000e41746c616e746120427261766573000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000d4e657720596f726b204d65747300000000000000000000000000000000000000",
  ],
  "statusCode": 200,
}
`;
